arch/riscv/core/offsets/offsets.c

```diff
@@ -13,9 +13,9 @@
  * structures.
  */
 
-#include <zephyr/arch/exception.h>
 #include <zephyr/kernel.h>
 #include <kernel_arch_data.h>
+#include <zephyr/arch/exception.h>
```

Comment:
Looks like a superfluous change?

Reply:
actually both exception.h and kernel_arch_data.h are not needed here. however arch should be included before exception to have the arch specific definitions.

Reply:
the problem arises that exceptions.h needs defintions  (`arch_esf->mepc` and `arch_esf->mstatu`s), which are now defined in arch depending on user or machine mode startup